lit_gpt/datasets/tinyllama.py

```diff
@@ -0,0 +1,99 @@
+        num_workers: The number of workers to use for the dataloaders.
+    """
+
+    def __init__(
+        self,
+        data_path: Union[str, Path] = Path("data/"),
+        seed: int = 42,
+        num_workers: int = 8,
+    ) -> None:
+        super().__init__()
+        self.data_path = data_path
+        self.seed = seed
+        self.num_workers = num_workers
+
+        self.batch_size = 1
+        self.seq_length = 2048
+
+    def connect(
+        self,
+        tokenizer: Optional[Tokenizer] = None,
+        batch_size: int = 1,
+        max_seq_length: Optional[int] = None
+    ) -> None:
+        self.batch_size = batch_size
+        self.seq_length = max_seq_length + 1  # Increase by one because we need the next token as well
+
+    def prepare_data(self) -> None:
+        pass
+
+    def train_dataloader(self) -> DataLoader:
+        from lightning.data.streaming import CombinedStreamingDataset, StreamingDataLoader, StreamingDataset, TokensLoader
+
+        # Could be a remote path (s3://) or a local path
+        slimpajama_train = str(self.data_path).rstrip("/") + "/slimpajama/train"
+        starcoder_train = str(self.data_path).rstrip("/") + "/starcoder"
+
+        train_datasets = [
+            StreamingDataset(
+                input_dir=slimpajama_train,
+                item_loader=TokensLoader(block_size=self.seq_length),
+                shuffle=True,
+                drop_last=True,
+            ),
+            StreamingDataset(
+                input_dir=starcoder_train,
+                item_loader=TokensLoader(block_size=self.seq_length),
+                shuffle=True,
+                drop_last=True,
+            ),
+        ]
+
+        # Mix SlimPajama data and Starcoder data with these proportions:
+        weights = (0.693584, 0.306416)
+        combined_dataset = CombinedStreamingDataset(datasets=train_datasets, seed=self.seed, weights=weights)
+        train_dataloader = StreamingDataLoader(
+            combined_dataset, batch_size=self.batch_size, pin_memory=True, num_workers=self.num_workers, drop_last=True
+        )
+        return train_dataloader
+
+    def val_dataloader(self) -> DataLoader:
+        from lightning.data.streaming import StreamingDataset, TokensLoader
+
+        slimpajama_val = str(self.data_path).rstrip("/") + "/slimpajama/val"
+
+        val_dataset = StreamingDataset(
+            input_dir=slimpajama_val,
+            item_loader=TokensLoader(block_size=self.seq_length),
+            shuffle=True,
+            # Consider setting to False, but we would lose some samples due to truncation when world size > 1
+            drop_last=True,
+        )
+        val_dataloader = DataLoader(
+            val_dataset, batch_size=self.batch_size, pin_memory=True, num_workers=self.num_workers, drop_last=True
+        )
+        return val_dataloader
+
+    def test_dataloader(self) -> DataLoader:
+        return self.val_dataloader()
```

Comment:
Since no script implements test, I wouldn't define this to avoid potential confusion